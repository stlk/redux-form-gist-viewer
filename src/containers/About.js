import React, { Component, PropTypes } from 'react';

export default () => <p>This is the about page</p>
