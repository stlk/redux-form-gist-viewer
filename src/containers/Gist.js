import React, { Component } from 'react';
import * as gistsActionCreator from 'redux/modules/gists';
import { connect } from 'react-redux';

import { GistForm, Profile, File } from 'components';

class Gist extends Component {
  componentDidMount() {
    this.props.loadGist(this.props.match.params.id);
  }

  render() {
    const { gists, gists: { editing } } = this.props;

    const gist = gists.data;
    if (!gists.loaded || !gist) return 'Loading...';

    return (
      <div>
        <Profile profile={gist.owner} />
        <button onClick={() => this.props.editStart(gist.id)}>Edit</button>
        {editing[gist.id] ? (
          <GistForm
            formKey={String(gist.id)}
            key={String(gist.id)}
            initialValues={gist}
          />
        ) : (
          <p>{gist.description}</p>
        )}
        {Object.keys(gist.files).map(filename => (
          <File key={filename} filename={filename} gist={gist} />
        ))}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    gists: state.gists,
  };
}

export default connect(mapStateToProps, {
  loadGist: gistsActionCreator.loadDetail,
  editStart: gistsActionCreator.editStart,
})(Gist);
