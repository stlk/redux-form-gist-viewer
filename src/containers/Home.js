import * as gistsActionCreator from '../redux/modules/gists';
import * as profileActionCreator from '../redux/modules/profile';
import { connect } from 'react-redux';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Profile from '../components/Profile';

export class Home extends Component {
  componentDidMount() {
    this.props.loadGists('stlk');
    this.props.loadProfile('stlk');
  }

  render() {
    const { gists, profile } = this.props;

    if (!gists.loaded) return 'Loading...';
    if (!profile.loaded) return 'Loading...';

    return (
      <div>
        <Profile profile={profile.data} />
        <ul>
          {gists.list.map(gist => (
            <li key={gist.id}>
              <Link to={`/gist/${gist.id}`}>{Object.keys(gist.files)[0]}</Link>
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    gists: state.gists,
    profile: state.profile,
  };
}

export default connect(mapStateToProps, {
  loadProfile: profileActionCreator.load,
  loadGists: gistsActionCreator.load,
})(Home);
