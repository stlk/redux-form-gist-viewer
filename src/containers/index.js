import NotFound from './About';
import About from './About';
import Home from './Home';

export { About, Home, NotFound }
