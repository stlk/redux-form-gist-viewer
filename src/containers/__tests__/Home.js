import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Home } from 'containers/Home';

it('renders without crashing', () => {
  var gists = {
    list: [
      {
        id: 'id',
        files: {
          'some.js': {},
        },
      },
    ],
    loaded: true,
  };
  var profile = {
    loaded: true,
    data: {
      name: 'joe doe',
      avatar_url: 'some/path',
    },
  };
  var result = renderer
    .create(
      <MemoryRouter>
        <Home
          gists={gists}
          profile={profile}
          loadGists={() => {}}
          loadProfile={() => {}}
        />
      </MemoryRouter>,
    )
    .toJSON();
  expect(result).toMatchSnapshot();
});
