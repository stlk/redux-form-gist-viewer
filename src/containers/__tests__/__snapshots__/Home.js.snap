// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="sc-bdVaJa fofobL"
  >
    <h1>
      joe doe
    </h1>
    <img
      alt=""
      src="some/path"
    />
  </div>
  <ul>
    <li>
      <a
        href="/gist/id"
        onClick={[Function]}
      >
        some.js
      </a>
    </li>
  </ul>
</div>
`;
